tests/test_worker.py

```diff
@@ -556,19 +594,20 @@ def test_behavior_called_when_execution_get_fails(
         job `run` is called, but the execution fails, there should be a delete
         call for that job if `keep_job` is False, and an exception should be raised.
         """
-        cloud_run_worker_job_config.keep_job = keep_job
-        mock_client.jobs().get().execute.return_value = self.job_ready
-        mock_client.jobs().run().execute.return_value = self.job_ready
-
-        # Getting execution will raise error
-        def raise_exception(*args, **kwargs):
-            raise Exception("This is an intentional exception")
-
-        monkeypatch.setattr("prefect_gcp.cloud_run.Execution.get", raise_exception)
-
-        with pytest.raises(Exception):
-            cloud_run_worker.run(flow_run, cloud_run_worker_job_config)
-        calls = list_mock_calls(mock_client, 2)
-        # breakpoint()
-        for call, expected_call in zip(calls, expected_calls):
-            assert call.startswith(expected_call)
+        async with CloudRunWorker("my-work-pool") as cloud_run_worker:
+            cloud_run_worker_job_config.keep_job = keep_job
+            mock_client.jobs().get().execute.return_value = self.job_ready
+            mock_client.jobs().run().execute.return_value = self.job_ready
+
+            # Getting execution will raise error
+            def raise_exception(*args, **kwargs):
+                raise Exception("This is an intentional exception")
+
+            monkeypatch.setattr("prefect_gcp.cloud_run.Execution.get", raise_exception)
+
+            with pytest.raises(Exception):
+                await cloud_run_worker.run(flow_run, cloud_run_worker_job_config)
+            calls = list_mock_calls(mock_client, 2)
+            # breakpoint()
```

Comment:
:)

Reply:
this is a faithful recreation of the test I copied over! (good catch, thank you 🙏)